src/plugins/telemetry/public/plugin.ts

```diff
@@ -91,6 +94,9 @@ export class TelemetryPlugin implements Plugin<TelemetryPluginSetup, TelemetryPl
       throw Error('Telemetry plugin failed to initialize properly.');
     }
 
+    this.canUserChangeSettings = this.getCanUserChangeSettings(application);
+    this.telemetryService.userCanChangeSettings = this.canUserChangeSettings;
```

Comment:
Why do we set `userCanChangeSettings` in the `telemetryService` from here as well as setting it from `config` in the [service](https://github.com/elastic/kibana/pull/76883/files#diff-38dfc0e43f1b1624252c66dcc2a8a7ebR102) itself?

Reply:
The main reason: the config is provided in the constructor but not this key because we don't have it at that moment (application.capabilities are not available yet), and we edit this parameter later on in the `setup` lifecycle step.
I could do `this.telemetryService.config = { ...this.telemetryService.config, userCanChangeSettings: this.canUserChangeSettings }` but it felt harder to read. That's why I created the specific getter/setter methods for `userCanChangeSettings` in the `telemetryService` class :)